frontend/src/Calendar/Agenda/AgendaEventConnector.js

```diff
@@ -6,6 +6,12 @@ import createQueueItemSelector from 'Store/Selectors/createQueueItemSelector';
 import createUISettingsSelector from 'Store/Selectors/createUISettingsSelector';
 import AgendaEvent from './AgendaEvent';
 
+const startDate = new Date();
+startDate.setDate(startDate.getDate() - 1);
+
+const endDate = new Date();
+endDate.setDate(endDate.getDate() + 31);
+
 function createMapStateToProps() {
   return createSelector(
     (state) => state.calendar.options,
```

Comment:
You can retrieve start and end from store, instead of calculating again here via

```
(state) => state.calendar.start,
(state) => state.calendar.end,
```